Feedback Second Implementation

Charpell opened the pull request:
### What does this PR do?
This PR implements all feedback gotten from the last defense session.

### Description of task completed?
+ The system functions works as expected. The system implements all endpoints and as well as front-end usage.
+ The "how to contribute section" of the readme file has been improved
+ The tests result has been improved and its description clearly states what the test is doing
+ File names have been renamed to match what is been exported. Variables and functions names have been renamed in camelCase.
+ The system now has an expiration time of 1 day, thus disallows a user to stay in for too long.

 
<img width="678" alt="31884983-e82c5978-b7e6-11e7-816d-5ddb8fae400a" src="https://user-images.githubusercontent.com/21138774/32659650-fb0fd2f8-c61f-11e7-93e5-1a678144fd60.png">

Charpell committed:
chore(code-refactoring): implement feedback

- installed express-validator to validate input field
- re-name files to follow airbnb standard
- refactored contribution write up in readme file

[Finished]

Charpell committed:
chore(fix-bug): implement feedback

- fixed issues with axios errors on the frontend

- create helper function to handle unresolved promise from axios

[Finished]

Charpell committed:
chore(code-refactoring): implement feedback

- write jsdoc for actions and api
- write descriptive custom error message instead of a generic one
- logs the user out after 24 hours
- change variable names to a descriptive name

[Finished]

Charpell committed:
chore(code-refactoring): implement feedback
 - fixed timeout on how long a user should be logged in

[finished]

Charpell committed:
chore(code-refactoring): implement feedback
- abstract bulky code into resubale component

[finished]

Charpell committed:
chore(code-refactoring): implement feedback
 - refactored navigation component into reuseable code

 [finished]

Charpell committed:
chore(code-refactoring): implement feedback
- Use Functions for all stateless component

[finished]

Charpell committed:
chore(code-refactoring): implement feedback
- Create 2 folders in components file; container and presentation
- Arrange all components into either container or presentation

[finished]

Charpell committed:
chore(code-refactor): implement feedback

- Write more component test
- Write elaborate jsDoc
- Use single responsibility principle in Message js

[finished]

Charpell committed:
chore(code-refactoring): implement feedback
 - Create middleware to validate users input and passed it to the route
- Create methods to handle email and sms notification

[finished]

Charpell committed:
test(client-test):refactor client test

- Write test for Store
- Configure Travis to run client test

 [finished]

Charpell committed:
chore(jsDoc): implement feedback
- Write JSDoc for AppStore

[finished]

Charpell committed:
Code Review

Charpell committed:
test(server): update server side test

- make server side test more descriptive
- server side return content of object
- update jsdoc
- reconfigure eslint settings
- refactor client files

[Finished]

Charpell committed:
chore(codecov): test coverage
- Generate test coverage report on codecov

[finished]

Charpell committed:
chore(controllers): refactor server controller files

- create middleware to check if a user is authenticated

- create helper function to save users who have read a message

- refactor server controller files to DRY codes

- create a function to catch client errors

[finished]

Charpell committed:
chore(test): update client side test

- update client side to and complete test
- install and configure jswebToken

[Finished ]

Charpell committed:
test(server): update server side test
- refactored every file for variable and function names

- write more concrete test for server side

[Finished ]

Seunadex commented on .eslintrc:
```diff
@@ -0,0 +1,105 @@
+    "jsx-a11y",
+    "import"
+  ]
+}
```
Kindly add a newline here and also do the same for most files here as they do not have a newline at end of file.

Charpell replied:
I have fixed this

Seunadex commented on client/src/js/components/container/Signup.jsx:
```diff
@@ -0,0 +1,210 @@
+     * @method onChange
+     *
+     * @description this method gets data from the store and sets to state
+		 *
```
The indentation for your comments looks off, kindly fix it and it occurs in several files too

Charpell replied:
I have fixed this

Seunadex commented on client/src/js/components/presentation/Navigation.jsx:
```diff
@@ -0,0 +1,30 @@
+const Navigation = () => (
+  <div>
+    <nav className="navbar navbar-inverse fixed-top"
+      style={{ borderRadius: '0px' }}>
```
Is there a reason for an inline styling here?

Charpell replied:
I have removed inline styling

Seunadex commented on client/src/js/tests/_mocks_/registerUserMock.json:
```diff
@@ -0,0 +1,10 @@
+      "redirectEventId": null
+  },
+  "myToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkaXNwbGF5TmFtZSI6IkhoIiwiZW1haWwiOiJkZGRAZ3QubmpnIiwiaWF0IjoxNTA5NjMyNDkwLCJleHAiOjE1MDk2MzYwOTB9.2JpqQzrNOwzSAMKOgaAhUAXlUhUF7yQ8fkEXNFZ9CTg"
+}
```
No newline at end of file.

Charpell replied:
I have fixed this

Charpell committed:
chore(code-refactor): implemnt feedback from PR

- removed inline styling
- added newline to end of files

[finished ]

Charpell committed:
chore(jsDoc): implement feedback

- Modify JsDoc template and styles

[finished ]

Charpell committed:
Merge pull request #44 from Charpell/Chore/fix-jsdoc

chore(jsDoc): implement feedback

Charpell committed:
Delete index.html

Charpell committed:
test(server): update server side test

- changed error message for authentication from specific  to generic error messages

- refactor test to only check for the value of a property instead of checking if it exist.

[finished ]

Charpell committed:
Delete index_bundle.js

Charpell committed:
Delete main.css

Charpell committed:
Delete .DS_Store

Charpell committed:
test(client-test): write store test

- write tests for functions in the AppStore

- create seeders file to abstract mock data

- remove old test snapshot

[finished ]

Charpell committed:
chore(code-refactor): implemnt feedback

- refactor Validate class

- ignore autogenerated files

- refactor Group Controller into Single responsibility principle

[finished ]

Charpell committed:
chore(code-climate): update travis.yml

- use code climate enviroment variable in travis

[finished ]

Charpell committed:
chore(change-filename): change the file name

- change filename in utils to AppAPI.js

Charpell committed:
chore(readme):update readme.md

- update the build status badge and coveralls badge to recent branch

[finished ]

Charpell committed:
Merge pull request #46 from Charpell/Chore/third-feedback-implementation

Feedback implementation
